src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,90 @@
-export const PeopleFilters = () => {
+import { Link, useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { useEffect } from 'react';
+import { Person } from '../types';
+import { getSearchWith } from '../utils/searchHelper';
+
+interface Props {
+  people: Person[];
+  setVisiblePeople: (arg: Person[]) => void;
+}
+
+enum FilterBySex {
+  'All' = '',
+  'Male' = 'm',
+  'Female' = 'f',
+}
+
+const centuriesForLinks = [16, 17, 18, 19, 20];
+
+function getFilteredPeople(
+  peopleWithParents: Person[],
+  filterSex: string,
+  query: string,
+  centuries: string[],
+) {
+  let filtered = [...peopleWithParents];
+
+  if (filterSex) {
+    filtered = filtered.filter(person => person.sex === filterSex);
+  }
+
+  if (query) {
+    const lower = query.toLowerCase().trim();
+
+    filtered = filtered.filter(
+      person =>
+        person.name.toLowerCase().includes(lower) ||
+        person.motherName?.toLowerCase().includes(lower) ||
+        person.fatherName?.toLowerCase().includes(lower),
+    );
+  }
+
+  if (centuries.length) {
+    filtered = filtered.filter(person => {
+      const birthCentury = Math.ceil(person.born / 100).toString();
+
+      return centuries.includes(birthCentury);
+    });
+  }
+
+  return filtered;
+}
+
+export const PeopleFilters = ({ people, setVisiblePeople }: Props) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const query = searchParams.get('query') || '';
+  const filterSex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  useEffect(() => {
+    const filtered = getFilteredPeople(people, filterSex, query, centuries);
+
+    setVisiblePeople(filtered);
+  }, [people, searchParams.toString()]);
```

Comment:
While this works, a better idea is to have the dependencies as the values you have used in the hook

```suggestion
  }, [people, filterSex, query, centuries);
```